processor/src/test/java/org/mapstruct/ap/test/bugs/_289/Issue289Mapper.java

```diff
@@ -0,0 +1,32 @@
+/**
+ *  Copyright 2012-2015 Gunnar Morling (http://www.gunnarmorling.de/)
+ *  and/or other contributors as indicated by the @authors tag. See the
+ *  copyright.txt file in the distribution for a full listing of all
+ *  contributors.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+package org.mapstruct.ap.test.bugs._289;
+
+import org.mapstruct.Mapper;
+import org.mapstruct.factory.Mappers;
+
+@Mapper
+public interface Issue289Mapper {
+
+    Issue289Mapper INSTANCE = Mappers.getMapper( Issue289Mapper.class );
+
+    Target sourceTarget(Source source);
```

Comment:
hmm. are you sure about the setter?.. the issue starts with:

"I created a mapping into a collection which is only accessible by its getter.
The generated mapper does a NULL check on the target read-only collection, but I would have expected a NULL check on the source collection too."

I'll add some additional test cases that you propose, but I reckon I wan tot keep this one as well.